### PR TITLE
ENG-10737: Fix for EE crash in full join query

### PR DESCRIPTION
**Pull request by wolffcm**
This fix is from Mike A.  His change was to update some of the logic in `TableTupleFilter` and to add a unit test that covers his change.

I found that the unit test didn't pass in memcheck mode, so fixed it.  I also added some comments to the class `TableTupleFilter`.

### CODE REVIEW

File: src/ee/storage/tabletuplefilter.h [hunk 1]
```diff
@@ -65,9 +65,15 @@ template<int8_t MARKER>
...
- * in the real table and the corresponding tuple index in the TableTupleFilter are related by the following
+ * A lightweight representation of a table - a contiguous array where
+ * each tuple (active and non-active) is represented as a byte with a
+ * certain value.  This lets clients of this class "tag" rows of a
+ * table with an 8-bit value.  For full outer joins, this is used to
+ * keep track of which tuples in the inner table were matched and
+ * which were not, so as to provide null-padded rows.
+ *
+ * The physical tuple address in the real table and the corresponding
+ * tuple index in the TableTupleFilter are related by the following
```
Comment (BillWhite) [inline src/ee/storage/tabletuplefilter.h]: An Ascii Art picture would really help here.

File: tests/ee/storage/tabletuplefilter_test.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    TempTable* getTempTable() {
+        return m_tempTable.get();
+    }
+
+
+private:
+
+    TupleSchema* createTupleScmema(int columns) {
+        std::vector<ValueType> all_types;
```
Comment (BillWhite) [inline tests/ee/storage/tabletuplefilter_test.cpp]: createTupleScmema -> createTupleSchema.

**Comment by BillWhite** [comment]
Makes sense to me.  I just had the one spelling correction.

**Comment by wolffcm** [comment]
Thanks Bill.  I wasn't sure if you were serious about the ASCII art.  I added some more clarifying comments though.

**Pull request merged**